src/patternfly/components/TextInputGroup/text-input-group.scss

```diff
@@ -162,8 +162,11 @@
 .pf-c-text-input-group__utilities {
   display: flex;
   align-items: center;
-  margin-right: var(--pf-c-text-input-group__utilities--MarginRight);
-  margin-left: var(--pf-c-text-input-group__utilities--MarginLeft);
+
+  &:not(:empty) {
```

Comment:
nice!! 🤩 This is great, but another way to do it that keeps the specificity lower would be to leave the existing declarations, then add `&:empty { --pf-c-text-input-group__utilities--MarginRight/Left: 0; }`. That might be a little easier to work with so we don't have to deal with specificity issues if we add other rules that need to modify the utilities margins.